executor/joiner.go

```diff
@@ -400,26 +363,29 @@ func (j *innerJoiner) tryToMatch(outer chunk.Row, inners chunk.Iterator, chk *ch
 	if inners.Len() == 0 {
 		return false, nil
 	}
-	j.chk.Reset()
-	chkForJoin := j.chk
-	if len(j.conditions) == 0 {
-		chkForJoin = chk
-	}
-	inner, numToAppend := inners.Current(), j.maxChunkSize-chk.NumRows()
-	for ; inner != inners.End() && numToAppend > 0; inner, numToAppend = inners.Next(), numToAppend-1 {
-		if j.outerIsRight {
-			j.makeJoinRowToChunk(chkForJoin, inner, outer)
-		} else {
-			j.makeJoinRowToChunk(chkForJoin, outer, inner)
+
+	return j.tryToMatchInnerAndOuter(j.outerIsRight, outer, inners, chk)
+}
+
+// tryToMatchInnerAndOuter does 2 things:
+// 1. Combine outer and inner row to join row.
+// 2. Check whether the join row matches the join conditions, if so, append it to the `outChk`.
+func (j *baseJoiner) tryToMatchInnerAndOuter(isRight bool, outer chunk.Row, inners chunk.Iterator, outChk *chunk.Chunk) (bool, error) {
```

Comment:
maybe we also can use this pattern in selectExec and other match then output execs as well if it works.

Reply:
Exactly, Nice catch!

Reply:
Maybe we should use an individual issue to trace and record the enhancement of the cache locality in the execution engine. But it's not easy to find the potential optimization opportunities without reading and analyzing the code. It may take us a long time to finish it.